mongofiles/mongofiles.go

```diff
@@ -234,49 +249,38 @@ func (mf *MongoFiles) getTargetGFSFiles() ([]*gfsFile, error) {
 	var gridFiles []*gfsFile
 	var err error
 
+	var query bson.M
+
 	// If mongofiles get ... is called, then query for all files
 	// specified in mf.FileNameList -- otherwise, preserve correct
 	// behavior for mongofiles get_id ...
 	if len(mf.FileNameList) > 0 {
-		query := bson.M{"filename": bson.M{"$in": mf.FileNameList}}
-
-		gridFiles, err = mf.findGFSFiles(query)
+		query = bson.M{"filename": bson.M{"$in": mf.FileNameList}}
+	} else if mf.FileNameRegex != "" {
+		query = bson.M{
+			"filename": bson.M{
+				"$regex":   mf.FileNameRegex,
+				"$options": mf.StorageOptions.RegexOptions,
+			},
+		}
+	} else if mf.Id != "" {
+		id, err := mf.parseOrCreateID()
 		if err != nil {
 			return nil, err
 		}
 
-		if len(gridFiles) < len(mf.FileNameList) {
-			return nil, fmt.Errorf("requested files not found: %v", mf.FileNameList)
-		}
+		query = bson.M{"_id": id}
 	} else {
-		var queryProp string
-		var query string
-
-		if mf.Id != "" {
-			queryProp = "_id"
-			query = mf.Id
-
-			id, err := mf.parseOrCreateID()
-			if err != nil {
-				return nil, err
-			}
-			gridFiles, err = mf.findGFSFiles(bson.M{"_id": id})
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			queryProp = "name"
-			query = mf.FileName
+		query = bson.M{"filename": mf.FileName}
+	}
 
-			gridFiles, err = mf.findGFSFiles(bson.M{"filename": mf.FileName})
-			if err != nil {
-				return nil, err
-			}
-		}
+	gridFiles, err = mf.findGFSFiles(query)
+	if err != nil {
+		return nil, err
+	}
 
-		if len(gridFiles) == 0 {
-			return nil, fmt.Errorf("no such file with %v: %v", queryProp, query)
-		}
+	if len(gridFiles) == 0 {
```

Comment:
This wasn't the condition for the first branch (`len(mf.FileNameList) > 0`). In that branch, we returned an error if `len(gridFiles) < len(mf.FileNameList)`. We can capture this in a general way by having this condition be `len(gridFiles) < someThreshold`. Perhaps we can default to `someThreshold := 1` (since that's the value for the latter two branches) and set `someThreshold = len(mf.FileNameList)` in the first branch?

It's up to you how to address this, the main thing is that this condition is different from the one before this change!

Reply:
Consider it done.